### PR TITLE
refactor: move exec metrics into executor

### LINKED ISSUE #9863: Live sync gas/s metric
### Describe the feature

Add panel for live sync gas/s, like exists for pipeline sync https://reth.paradigm.xyz/d/2k8BXz24x/reth?orgId=1&refresh=30s&from=1722251367510&to=1722254967510&viewPanel=56.

### Additional context

_No response_

### LINKED ISSUE #10397: Track gas used metrics in executor directly
### Describe the feature

Currently we only track how much gas we've executed here: https://github.com/paradigmxyz/reth/blob/5470574bf208b36696b44a4ecd591cfb3cc98496/crates/stages/api/src/metrics/sync_metrics.rs#L35-L41

This only works for the execution stage (i.e. backfill sync). We should move this directly into the executor so it also works for live sync (ref #9863).

Additionally, we should expose an `mgas_per_second` metric that should expose how much gas per second was processed. For example, a 10M gas block processed in 10 seconds would set this value to 1.

This should also be reflected in the Grafana dashboard.

### Additional context

Currently, we calculate mgas/s in the metrics dashboard using Prometheus, but doing this for live sync would be misleading, since it would divide the gas processed over the entire 12 second slot time.

### PR DESCRIPTION
**Pull request by onbjerg**
Removes `ExecutionStageMetrics` and instead adds an `ExecutorMetrics` used to track things like gas processed, and more in the future.

Currently is not integrated into the blockchain tree.

This PR also adds an instantaneous gas/s metric to make this work well with live sync -- currently, if we just added the total gas processed from live sync and did a `rate` in Prometheus over it, performance would look abysmal on live sync, since it would divide the gas processed over the entire slot time (12s).

I've implemented this with a `metered` fn that takes a closure and measures the relevant metrics from the input and output. This is ok except it does not handle generic inputs.

An alternative I've considered (and still am) was adding some metering to the executor traits instead, since it allows us to handle generic inputs and outputs more easily

Future work could include adding metrics for sloads/sstores, bytecode reads/writes, and account loads/updates

Closes #10397 and closes #9863

### CODE REVIEW

**Comment by onbjerg** [comment]
Last task is to update grafana

File: crates/stages/api/src/metrics/listener.rs [hunk 1]
```diff
@@ -82,9 +76,6 @@ impl MetricsListener {
                     stage_metrics.entities_total.set(total as f64);
                 }
             }
-            MetricEvent::ExecutionStageGas { gas } => {
-                self.sync_metrics.execution_stage.mgas_processed_total.increment(gas / MEGAGAS)
-            }
```
Comment (mattsse) [inline crates/stages/api/src/metrics/listener.rs:87]: yeah this was a bit weird

File: crates/evm/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    /// The instantaneous amount of gas processed per second.
+    pub gas_per_second: Gauge,
+}
+
+impl ExecutorMetrics {
+    /// Execute the given block and update metrics for the execution.
+    pub fn metered<F, R>(&self, input: BlockExecutionInput<'_, BlockWithSenders>, f: F) -> R
+    where
+        F: FnOnce(BlockExecutionInput<'_, BlockWithSenders>) -> R,
+    {
```
Comment (mattsse) [inline crates/evm/src/metrics.rs:28]: this seems fine

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**